test_word_search.py

```diff
@@ -0,0 +1,73 @@
+"""
+This module contains unit tests for word search solution
+"""
+from word_search import get_valid_words, generate_grid, search_word
+
+FILE_NAME = 'words_dict.txt'
+WORDS = get_valid_words(FILE_NAME)
+GRID_SIZE = 15
+
+
+def test_valid_words_amount():
+    assert len(get_valid_words(FILE_NAME)) == 62798
+
+
+def test_grid_size():
+    grid = generate_grid(GRID_SIZE)
+    assert len(grid) == GRID_SIZE
+    for line in grid:
+        assert len(line) == GRID_SIZE
+
+
+def test_word_search_1():
+    grid = [['p', 'h', 'o'],
+            ['j', 'e', 'y'],
+            ['r', 'q', 't']]
+
+    words = ['pet', 'ho', 'he', 'oh', 'eh', 'ye', 'yo', 're']
+
+    assert list(search_word(grid, WORDS)) == words
+
+
+def test_word_search_2():
+    grid = [['a', 'b', 'c', 'l', 'y'],
+            ['b', 'h', 'c', 'd', 'm'],
+            ['c', 'g', 'd', 'i', 'a'],
+            ['d', 'i', 'k', 'm', 'x'],
+            ['m', 'e', 'o', 'w', 'r']]
+
+    words = ['ah', 'ha', 'dim', 'ma', 'my', 'mi', 'mike', 'dim', 'id', 'id', 'aid', 'ax', 'am', 'am', 'ad',
+             'id', 'id', 'kid', 'mi', 'mid', 'ma', 'me', 'meow', 'mi', 'mid', 'middy', 'em', 'ow', 'woe']
+
+    assert list(search_word(grid, WORDS)) == words
+
+
+def test_word_search_3():
+    grid = [['o', 'd', 'p', 'p', 'x', 'w', 'c', 'p', 'w', 'j', 'z', 'i', 'd', 'y', 't'],
+            ['j', 'n', 'x', 'g', 'p', 'x', 'w', 'c', 'a', 'i', 'q', 'j', 'k', 'n', 'r'],
+            ['e', 'h', 'i', 'c', 'c', 'x', 'g', 'j', 'f', 'p', 'f', 'x', 'm', 'p', 'k'],
+            ['q', 'm', 'u', 'c', 'p', 'e', 'm', 'p', 'c', 'm', 'v', 'a', 'x', 'p', 'v'],
+            ['j', 'v', 'z', 'v', 'q', 'y', 'w', 'j', 'e', 'v', 'm', 'n', 'l', 'i', 'n'],
+            ['z', 'b', 'e', 'x', 'd', 'g', 'p', 'j', 'f', 'y', 't', 'c', 'l', 'i', 'f'],
+            ['l', 'f', 'x', 'h', 'x', 'p', 'x', 'f', 'i', 'p', 'q', 'q', 'v', 'f', 'f'],
+            ['d', 'm', 'n', 'b', 'm', 'n', 'i', 'h', 'w', 'e', 'i', 'l', 'q', 'q', 'g'],
+            ['i', 't', 'r', 'l', 't', 'c', 'g', 'm', 'g', 'b', 'q', 'c', 'u', 'q', 'i'],
+            ['z', 'x', 'c', 'p', 'u', 'v', 'o', 'i', 's', 'a', 'w', 'n', 'n', 'u', 'v'],
+            ['l', 'c', 'i', 'k', 'v', 'v', 't', 'e', 'z', 'c', 'v', 'v', 'd', 'c', 'j'],
+            ['f', 'd', 'u', 'z', 'c', 'v', 'f', 'r', 'g', 'c', 'x', 'u', 'x', 'y', 'u'],
+            ['h', 'p', 'n', 'h', 'p', 'f', 'l', 'j', 'c', 'k', 'j', 'q', 'h', 'e', 'y'],
+            ['z', 'e', 'k', 'a', 'o', 'w', 'l', 'm', 'y', 'b', 'v', 'r', 'h', 'u', 'y'],
+            ['t', 'c', 'g', 'v', 'g', 'd', 'd', 'p', 'j', 'x', 'z', 'c', 'k', 'r', 'b']]
+
+    words = ['on', 'do', 'pa', 'pap', 'id', 'no', 'if', 'if', 'eh', 'em', 'hi', 'he', 'in', 'fa', 'pi', 'pa',
```

Comment:
How do you know that this set of words is correct?

Reply:
I got it with search_word function